index.bs

```diff
@@ -843,4 +1380,227 @@ result = (
   unknown-error: 199
 )
 
+; type key 17
+remote-playback-availability-request = {
+  request
+  1: [* text] ; urls
+  2: microseconds ; watch-duration
+  3: int ; watch-id
+}
+
+; type key 18
+remote-playback-availability-response = {
+  response
+  url-availabilities: [* url-availability]
+}
+
+; type key 114
+remote-playback-availability-event = {
+  1: int ; watch-id
+  2: [* url-availability] ; url-availabilities
+}
+
+; type key 115
+remote-playback-start-request = {
+  request
+  1: remote-playback-id ; remote-playback-id
+  2: [* text] ; urls
+  ? 3: [* text] ; text-track-urls
+  ? 4: [* http-header] ; headers
+  ? 5: remote-playback-controls ; controls
+}
+
+; type key 116
+remote-playback-start-response = {
+  response
+  1: &result ; result
+  ? 2: remote-playback-state ; state
+}
+
+; type key 117
+remote-playback-termination-request = {
+  request
+  1: remote-playback-id ; remote-playback-id
+  2: &(
+    user-terminated-via-controller: 11
+    unknown: 255
+  ) ; reason
+}
+
+; type key 118
+remote-playback-termination-response = {
+  response
+  1: &result ; result
+}
+
+; type key 119
+remote-playback-termination-event = {
+ 1: remote-playback-id ; remote-playback-id
+ 2: &(
+    receiver-called-terminate: 1
+    user-terminated-via-receiver: 2
+    receiver-idle-too-long: 30
+    receiver-powering-down: 100
+    receiver-crashed: 101
+    unknown: 255
+  ) ; reason
+}
+
+; type key 19
+remote-playback-modify-request = {
+  request
+  1: remote-playback-id ; remote-playback-id
+  2: remote-playback-controls ; controls
+)
+
+; type key 20
+remote-playback-modify-response = {
+  response
+  1: &result ; result
+  ? 2: remote-playback-state ; state
+}
+
+; type key 21
+remote-playback-state-event = {
+  1: remote-playback-id ; remote-playback-id
+  2: remote-playback-state ; state
+}
+
+remote-playback-id = uint
+
+remote-playback-controls = {
+  ? 1: url ; source
+  ? 2: &(none, metadata, auto) ; preload
+  ? 3: bool ; loop
+  ? 4: bool ; paused
+  ? 5: bool ; muted
+  ? 6: float ; volume
+  ? 7: media-time ; seek
+  ? 8: media-time ; fast-seek
+  ? 9: float ; playback-rate
+  ? 10: url ; poster
+  ? 11: [* text] ; enabled-audio-track-ids
+  ? 12: text ; selected-video-track-id
+  ? 13: [* added-text-track] ; added-text-tracks
+  ? 14: [* changed-text-track] ; changed-text-tracks
+}
+
+remote-playback-state = {
+  ? 1: (rate: bool, preload: bool, poster: bool, added-text-track: bool, added-cues: bool) ; supports
+
+  ? 2: url ; source
+  ? 3: &(
+    empty: 0
+    idle: 1
+    loading: 2
+    no-source: 3
+  ) ; loading
+  ? 4: &(
+    nothing: 0
+    metdata: 1
+    current: 2
+    future: 3
+    enough: 4
+  ) ; loaded
```

Comment:
Even if `3` and `4`'s types aren't used elsewhere, I think pulling them out into enums defined outside this map would be more readable. Plus the generated C++ code would be more readable and easier to use